File: julia/roman-numerals.jl
const romans = [
    (1000, "M"),
    (900 , "CM"),
    (500 , "D"),
    (400 , "CD"),
    (100 , "C"),
    (90 , "XC"),
    (50 , "L"),
    (40 , "XL"),
    (10 , "X"),
    (9 , "IX"),
    (5 , "V"),
    (4 , "IV"),
    (1 , "I")
]

function to_roman(number)
    if number <= 0
        error()
    end

    s = ""
    for (k, v) in romans
        while number >= k
            s *= v
            number -= k
        end
    end
    s
end

File: dart/roman-numerals.dart
extension ConvertToRomanNumerals on int {
  String toRoman() {
    final romans = [
      "M",
      "CM",
      "D",
      "CD",
      "C",
      "XC",
      "L",
      "XL",
      "X",
      "IX",
      "V",
      "IV",
      "I"
    ];
    final nums = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
    String s = "";
    int n = this;
    for (int i = 0; i < romans.length; i++) {
      while (n >= nums[i]) {
        s += romans[i];
        n -= nums[i];
      }
    }
    return s;
  }
}
